examples/evaluation_and_profiling/swe_bench/src/nat_swe_bench/predictors/predict_iterative/predict_iterative.py

```
@@ -0,0 +1,505 @@
+        except (TimeoutError, subprocess.TimeoutExpired) as e:
+            # Extract output from exception if available (only subprocess.TimeoutExpired has output attribute)
+            if isinstance(e, subprocess.TimeoutExpired) and hasattr(e, "output") and e.output:
+                output = e.output.decode("utf-8", errors="replace")
+            else:
+                output = ""
+            # Format timeout message using template
+            timeout_message = self._TIMEOUT_TEMPLATE.format(
+                action=command,
+                output=output
+            )
+            raise ExecutionTimeoutError(timeout_message)
```

Ruff (0.14.11)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)